File: app/src/main/java/com/spiderbiggen/randomchampionselector/ui/activities/ButtonActivity.java
package com.spiderbiggen.randomchampionselector.ui.activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v4.app.ActivityOptionsCompat;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;

import com.spiderbiggen.randomchampionselector.R;

/**
 * Fullscreen activity
 * Created by Stefan on 9-5-2015.
 */
public abstract class ButtonActivity extends AppCompatActivity {

    private static final String TAG = ButtonActivity.class.getSimpleName();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        super.onCreate(savedInstanceState);
    }

    @Override
    public final boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.menu_scrolling, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                supportFinishAfterTransition();
                return true;
            case R.id.action_settings:
                startActivity(new Intent(this, SettingsActivity.class));
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    public void openChampion(View view) {
        View fab = findViewById(R.id.fab);
        Intent intent = getChampionIntent();
        startActivity(intent);
    }

    protected Intent getChampionIntent() {
        return new Intent(this, ChampionActivity.class);
    }

    protected Intent getChampionListIntent() {
        return new Intent(this, ListChampionsActivity.class);
    }

    @Override
    public void startActivity(Intent intent) {
        ActivityOptionsCompat options = ActivityOptionsCompat.makeCustomAnimation(this, android.R.anim.fade_in, android.R.anim.fade_out);
        super.startActivity(intent, options.toBundle());
    }
}

File: app/src/main/java/com/spiderbiggen/randomchampionselector/activities/ButtonActivity.kt
package com.spiderbiggen.randomchampionselector.activities

import android.content.Intent
import android.support.v4.app.ActivityOptionsCompat
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.View

import com.spiderbiggen.randomchampionselector.R

/**
 * Fullscreen activity Created by Stefan on 9-5-2015.
 */
abstract class ButtonActivity : AppCompatActivity() {

    protected val championIntent: Intent
        get() = Intent(this, ChampionActivity::class.java)

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean =
            when (item.itemId) {
                android.R.id.home -> {
                    supportFinishAfterTransition()
                    true
                }
                R.id.action_settings -> {
                    startActivity(Intent(this, SettingsActivity::class.java))
                    true
                }
                R.id.action_force_refresh -> {
                    startActivity(LoaderActivity.createStartIntent(this, true))
                    true
                }
                else -> super.onOptionsItemSelected(item)
            }

    open fun openChampion(view: View) {
        startActivity(championIntent)
    }

    override fun startActivity(intent: Intent) {
        val options = ActivityOptionsCompat
                .makeCustomAnimation(this, android.R.anim.fade_in, android.R.anim.fade_out)
        super.startActivity(intent, options.toBundle())
    }
}
